lib/os/onoff.c

```diff
@@ -40,7 +129,8 @@ static int validate_args(const struct onoff_manager *mgr,
 	int rv = sys_notify_validate(&cli->notify);
```

Comment:
While browsing the onoff.c, I noticed that the code does not follow the coding guideline but introduces variables in the middle of the blocks. Would it be possible to fix these?

Reply:
Sorry to be obtuse; can you point me to the relevant guideline?

Mid-block declarations have been legal in C for twenty years, in various discussions I remember consensus that C99 was assumed to be the minimum supported versions.  Though I don't think anybody documented that.

Reply:
Of course compiler supports this just fine. We are following linux coding style in zephyr.

Reply:
I don't see anything about this in https://www.kernel.org/doc/html/latest/process/coding-style.html

There's good reason to define the variable at the point where you can assign it an initial value, which is what I've done.  So while it would be possible to "fix" these, I don't plan to do so.  You may block the PR if you feel strongly enough.

Personally I wish we could get some traction on #21392 so we could stop having these arguments and instead have an objective standard for acceptability.

Reply:
We already rely on too many C99 (and GNU99) features to deny its existence and effectiveness, like some cave-residing individuals do.

I also believe mid-function variable declaration, when it makes sense, is helpful and should even be preferred if it results in a tidier/easier-to-follow code.

In fact, mid-function beginning-of-scope variable declarations are widely preferred in the Linux kernel, although that is arguably different to what is being done here.